File: Project-Films-Angular/src/app/movie.ts
export enum MovieCategory {
  DRAMA = ('DRAMA'),
  COMEDY = ('COMEDY'),
  FAMILY = ('FAMILY'),
  ACTION = ('ACTION'),
  SCI_FI = ('SCI-FI'),
  CLASSIC = ('CLASSIC')
}

export class Movie {
  public id: number = undefined;
  public title: string = undefined;
  public category: MovieCategory = undefined;
  public year: string = undefined;
  public cast: string = undefined;
  public director: string = undefined;
  public story: string = undefined;
  public price: number = undefined;
}

File: Project-Films-Spring/src/main/java/put/take/persistance/Movie.java
package put.take.persistance;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "movies")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class Movie implements Serializable, Cloneable {

    @Id
    private Long id;
    private String title;

    private String category;
    private String year;
    private String cast;
    private String director;
    private String story;
    private Double price;

    @Override
    public Movie clone() throws CloneNotSupportedException {
        return (Movie)super.clone();
    }
}
